### PR TITLE
java, update DPG how-to guide

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Fill in the Java part of how-to guide.

A few scenario is not supported right now, comment as such.

### CODE REVIEW

File: docs/howtos/DataPlane Generation - DPG/05union.mdx [hunk 1]
```diff
@@ -73,10 +88,12 @@ you must use `@sharedRoute`. By default, codegen will generate one method per op
...
-op a(x: int): float;
+@sharedRoute
+@route("/foo")
+op a(x: int32): float;
 
-@sharedRoute("/foo")
+@sharedRoute
+@route("/foo")
 op b(x: string): int64;
 ```
```
Comment (weidongxu-microsoft) [inline docs/howtos/DataPlane Generation - DPG/05union.mdx:98]: Update this typespec. The origin one is not valid.

https://typespec.io/docs/libraries/http/reference/decorators#@TypeSpec.Http.sharedRoute

File: docs/howtos/DataPlane Generation - DPG/03convenient.mdx [hunk 2]
```diff
@@ -174,7 +185,16 @@ FIXME
 <TabItem value="java" label="Java" >
 
 ```java
-FIXME
+package com.petstorenamespace.implementation.models;
```
Comment (weidongxu-microsoft) [inline docs/howtos/DataPlane Generation - DPG/03convenient.mdx:188]: case of internal in Java

model in impl, which would not get exposed to customer

Comment (haolingdong-msft) [reply docs/howtos/DataPlane Generation - DPG/03convenient.mdx:188]: nit: add a comment to this code? saying the model is in `impl` package.

File: docs/howtos/DataPlane Generation - DPG/03convenient.mdx [hunk 3]
```diff
@@ -235,7 +255,20 @@ FIXME
 <TabItem value="java" label="Java" >
 
 ```java
-FIXME
+package com.azure.ai.openai.models;
+
+public final class AzureCognitiveSearchIndexFieldMappingOptions {
+    public AzureCognitiveSearchIndexFieldMappingOptions()
```
Comment (haolingdong-msft) [inline docs/howtos/DataPlane Generation - DPG/03convenient.mdx:261]: What is the difference with not having usage? mention using comments?

Comment (weidongxu-microsoft) [reply docs/howtos/DataPlane Generation - DPG/03convenient.mdx:261]: usually the public/private of the ctor.

If it is output only, its ctor is private.

File: docs/howtos/DataPlane Generation - DPG/03convenient.mdx (hunk 2 above)
Comment (weidongxu-microsoft) [reply docs/howtos/DataPlane Generation - DPG/03convenient.mdx:194]: good point, added.

File: docs/howtos/DataPlane Generation - DPG/03convenient.mdx [hunk 4]
```diff
@@ -174,7 +188,19 @@ FIXME
...
+
+package com.petstorenamespace.implementation.models;
+public final class OutputModel {
+    public String getName()
+}
+
+package com.petstorenamespace;
+public final class PetStoreNamespaceClient {
+    Response<BinaryData> getModelWithResponse(String name, RequestOptions requestOptions)
+    OutputModel getModel(String name)
```
Comment (XiaofeiCao) [inline docs/howtos/DataPlane Generation - DPG/03convenient.mdx:202]: nit, mention `//protocol`, `//convenience`?

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**